ADR/0029-basic-authentication-secret.md

```diff
@@ -0,0 +1,88 @@
+# 29. Basic Authentication Secret
+
+Date: 2023-08-25
+
+## Status
+
+Proposed
+
+## Context
+
+- As a user, I need a secure way to configure sensitive information/tokens/credentials for various components within the RHTAP environment. This includes providing credentials for accessing source code repositories on GitHub and other authentication needs.
+- As a user, I expect to have a place where I can list already provided credentials and adjust them it if needed.
+
+## Decision
+
+### Persisting Sensitive Information
+
+We propose that the UI captures sensitive information through form input and creates `RemoteSecret` Custom Resources (CRs) to store this data. The `RemoteSecret.data` will be intercepted by a webhook, securely stored, and removed from the `RemoteSecret` CR to ensure that data is not kept in etcd.
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: github-com-token
+    namespace: myns
+    labels:
+        appstudio.redhat.com/sp.host: github.com
+spec:
+    secret:
+        type: kubernetes.io/basic-auth
+data:
+  username: Z2VuYQ==
+  password: Z2VuYQ==
+```
+### Accessing Sensitive Information
+
+1. RHTAP components will search for `RemoteSecret` instances based on the `appstudio.redhat.com/sp.host` label.
+2. For the required `RemoteSecret`s, the components will expand the targets using the desired secret name, label, and annotations to access the secret data.
+3. The controller will retrieve `secretData` from permanent storage and deliver it to the desired location.
+4. Once the secret is no longer needed, the components will remove the `RemoteSecret`'s target.
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: github-com-token
+    namespace: myns
+    labels:
+        appstudio.redhat.com/sp.host: github.com
+spec:
+    targets:
+    - namespace: myns
+      secret:
+        name: tekton-secret
+        annotations:
+            tekton.dev/git-0: https://github.com
+    secret:
+        type: kubernetes.io/basic-auth
+
+```
+### Fine-Grained Credentials
+
+Users can specify resource names through the UI, which will be added as comma-separated annotation `appstudio.redhat.com/sp.repository` to the `RemoteSecret` CR. RHTAP components will consider these annotations during `RemoteSecret` searches.
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: github-com-token
+    namespace: myns
+    annotations:
+        appstudio.redhat.com/sp.repository:   "redhat/secret, redhat/opera"
```

Comment:
```suggestion
        appstudio.redhat.com/sp.repository: "redhat/secret,redhat/opera"
```
I believe these should be separated with a comma without the whitespace. If you think we should take the whitespace into consideration let me know.